openmdao/core/system.py

```diff
@@ -1658,8 +1657,77 @@ def _setup_relevance(self, mode, relevant=None):
         relevant : dict or None
             Dictionary mapping VOI name to all variables necessary for computing
             derivatives between the VOI and all other VOIs.
-
         """
+        # Update driver quantities to apply unit conversions.
```

Comment:
This setup of unit conversions doesn't really belong in the `setup_relevance` method.  Could this stuff be done in the driver after it collects the desvars/constraints/objective?

Reply:
Moved that stuff into its own function and called it from the first method in driver that is called under final_setup.